### DISCUSSION TITLE
About Terminating Solver

### DISCUSSION THREAD

**Discussion by Ajay3oodles** [TimefoldAI/timefold-solver#1199] [General]
I have used this to terminate my solving process but I have noticed that even after this my solving continues and then after some time it eventually stops
 solverManager.terminateEarly(Long.valueOf(uniqueOrgId));

![Screenshot from 2024-11-12 16-01-44](https://github.com/user-attachments/assets/b1f65071-c166-4574-8ada-7c7a8bbf751c)
![Screenshot from 2024-11-12 16-00-29](https://github.com/user-attachments/assets/e172a1ae-cf16-4f6f-aaeb-58049225ec45)

It continued in separate thread and then stopped

**Comment by Christopher-Chianelli** [comment]
I see you are using a Custom Phase. The solver terminates when it is first able to. For a Local Search/Construction Heuristic/Exhaustive Search phase, that is after evaluation of a move. For a Custom Phase, it is checked after a `CustomPhaseCommand` is executed. Depending on how long the `CustomPhaseCommand` takes, there might be a long delay between request for termination and actual termination.

**Reply by Christopher-Chianelli** [reply]
Created https://github.com/TimefoldAI/timefold-solver/issues/1200 as a RFE to allow `CustomPhaseCommand` to check if the Solver was terminated early.

**Reply by Ajay3oodles** [reply]
Thank you so much for you response,really appreciated.
